src/main/scala/io/iohk/ethereum/network/p2p/messages/CommonMessages.scala

```diff
@@ -22,12 +22,20 @@ object CommonMessages {
 
       override def toRLPEncodable: RLPEncodeable = {
         import msg._
-        msg match {
-          case _: Status63 =>
-            RLPList(protocolVersion, networkId, totalDifficulty, bestHash, genesisHash)
+        msg.code match {
+          case `code63` =>
+            RLPList(protocolVersion, networkId, chainWeight.totalDifficulty, bestHash, genesisHash)
 
-          case _: Status64 =>
-            RLPList(protocolVersion, networkId, totalDifficulty, latestCheckpointNumber, bestHash, genesisHash)
+          case `code64` =>
+            RLPList(
+              protocolVersion,
+              networkId,
+              //TODO: should it be an RLPList? Should we define encoding in ChainWeight object?
```

Comment:
👍  for encoding in ChainWeight object but move it to the end or merge both RLPLists into one

Reply:
I'm not yet sure how to handle that while keeping the compatibility with v63 messages